train_fromRoot.py

```diff
@@ -0,0 +1,197 @@
+# Import libraries
+import tensorflow
+import tensorflow.keras.backend as K
+from tensorflow.keras import optimizers, initializers
+from tensorflow.keras.callbacks import ReduceLROnPlateau, ModelCheckpoint, EarlyStopping, CSVLogger
+from tensorflow.keras.utils import plot_model
+from tensorflow.keras.models import Model
+from sklearn.model_selection import train_test_split
+
+import numpy as np
+import tables
+import matplotlib.pyplot as plt
+import argparse
+import math
+#import setGPU
+import time
+import os
+import pathlib
+import datetime
+import tqdm
+import h5py
+
+#Import custom modules
+
+from Write_MET_binned_histogram import *
+from cyclical_learning_rate import CyclicLR
+from models import *
+from utils import *
+from loss import custom_loss
+#from epoch_all import epoch_all
+from DataGenerator import DataGenerator
+
+def main(args):
```

Comment:
there's a lot of code duplication between this `main` function and the `train_fromh5` `main` function. this should be avoided (if there's a bug, you only have to fix it one place).

also it might be better to unify all the code into a single `train.py` file

the way i might structure the code is like this:

```
import whatever

def prepareArrays():
  ...
def prepareGenerators():
  ...
def prepareCallbacks():
  ...
def train():
  ...
def test():
  ...
  
def main(args):
  if args.dataSetType == 'h5': prepareArrays()
  else: prepareGenerators()
  prepareCallbacks()
  train()
  test()
  
if __name__ == "__main__":
  args  = ...
  main(args)
```

or something like that